core/src/main/java/org/apache/brooklyn/core/mgmt/ha/OsgiArchiveInstaller.java

```diff
@@ -60,6 +60,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.google.common.base.Function;
```

Comment:
I only just noticed that a bunch of this code is in `core/mgmt/ha`. Why is that? I'd have thought it would equally apply to non-ha (e.g. persisting your state locally, and then restarting Brooklyn).

Reply:
indeed, longstanding misplacement.  would be good to move `OsgiManager` and the related classes.